perl-lib/OESS/lib/OESS/MPLS/Discovery/Interface.pm

```diff
@@ -35,8 +35,59 @@ sub new{
 sub process_results{
     my $self = shift;
     my %params = @_;
+    my $node_name = $params{'node'};
+    my $interfaces = $params{'interfaces'};
+    foreach my $interface (@$interfaces) {
+	my $interface_id = $self->{'db'}->get_interface_id_by_names(node => $node_name, interface => $interface->{'name'});
+	if (!defined($interface_id)) {
+	    $self->{'db'}->_start_transaction();
```

Comment:
So I really want to wrap the entire processing in a transaction... so We should move the _start_transaction to before the foreach statement.... and the commit to after it